### PR TITLE
Add `core.blocks`

### LINKED ISSUE #2: Table: Blocks
**Description**
Let's create a Blocks table using `chainwalkers.prod.harmony_blocks` as the source data table.

**To Do Next:**

- [x] Design the schema for this table
- [x] Get confirmation on schema
- [x] Build schema
- [x] Schema build review
- [x] Submit Pull Request

### PR DESCRIPTION
**Pull request by chuxinh**
## What's change
- Update `block.sql` following the schema design: https://github.com/MetricsDAO/harmony_dbt/issues/2#issuecomment-998369924
- Add the blocks yml update to schema. However, I'm wondering if we should rather delete the `schema.yml` and create yml for each table. i.e. `block.yml`, `tx.yml` etc? 

## Testing
```
dbt run --select core/blocks.sql
Running with dbt=0.21.1
Found 3 models, 16 tests, 0 snapshots, 0 analyses, 175 macros, 0 operations, 0 seed files, 2 sources, 0 exposures
The selection criterion 'core/blocks.sql' does not match any nodes

WARNING: Nothing to do. Try checking your model configs and model specification args
root@ae102a794681:/harmony# dbt run --select core.blocks
Running with dbt=0.21.1
Found 3 models, 16 tests, 0 snapshots, 0 analyses, 175 macros, 0 operations, 0 seed files, 2 sources, 0 exposures

00:16:23 | Concurrency: 4 threads (target='dev')
00:16:23 | 
00:16:23 | 1 of 1 START incremental model DEV.blocks............................ [RUN]
00:16:34 | 1 of 1 OK created incremental model DEV.blocks....................... [SUCCESS 1 in 11.00s]
00:16:34 | 
00:16:34 | Finished running 1 incremental model in 17.90s.

Completed successfully

Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1](https://github.com/MetricsDAO/harmony_dbt/issues/2#issuecomment-998369924)
```

`dev.blocks`
image

### CODE REVIEW

File: models/core/blocks.sql
Comment (antonyip) [inline models/core/blocks.sql]: Might want to cluster_by block_timestamp? Maybe snowflake does it for us automatically tho?

Comment (chuxinh) [reply models/core/blocks.sql]: oh i actually didn't know this before but rather order the table by block_timestamp in the last line.
I think they do the same thing: https://docs.getdbt.com/reference/resource-configs/snowflake-configs
but happy to update specifying `cluster_by` in the config. what do you think?

Comment (antonyip) [reply models/core/blocks.sql]: Sounds good. I've just been crawling flipside's pull requests and they all seem to use cluster_by. Might be good to follow them for now since they've been doing the same things for longer.

File: models/core/blocks.sql
Comment (antonyip) [inline models/core/blocks.sql:34]: I think the where clause can go into the CTE? Should be less load on the server..

Comment (forgxyz) [reply models/core/blocks.sql:34]: I did that based on [Tristan Handy's commentary here](https://discourse.getdbt.com/t/why-the-fishtown-sql-style-guide-uses-so-many-ctes/1091). Evidently, Snowflake treats the SELECT * as a pass-through, so there's actually no negative impact on performance.

File: models/core/txs.yml
Comment (chuxinh) [inline models/core/txs.yml]: probably can leave out the tx one given @mattromano is working on it in a separate PR

File: models/core/blocks.sql
Comment (antonyip) [reply models/core/blocks.sql:34]: Welp, I got schooled :D

File: models/core/txs.yml
Comment (forgxyz) [reply models/core/txs.yml]: Agree, I'll delete that

**Review by forgxyz — APPROVED** [review]
Schema design and implementation look good to me, per discussions.

**Comment by chuxinh** [comment]
pending @jfmyers's approval as well as dropping the existing table in prod for this to run full rebuild

**Comment by drakedanner** [comment]
@chuxinh you can merge this!

Per @jfmyers :

it's setup now so that any PR to main requires 1 other person to review and approve it, once approved the PR creator can merge

**Pull request merged**
